Problem5/src/problem5/surveyWindow.java

```diff
@@ -0,0 +1,349 @@
+        
+        //Retrieves data from the current file
+        int surveyCount = retrieve.surveyCount(filename);
+        int[] question1 = retrieve.question1Data(filename);
+        int[] question2 = retrieve.question2Data(filename); 
+        String[] question3 = retrieve.question3Data(filename, surveyCount);
+        
+        //Updates the file
+        file.update(filename, surveyCount, question1, question2, question3, getCheckBox(), getComboBox(), getTextArea());
+    }//GEN-LAST:event_submitButtonMouseClicked
+
+    /**
+     * @param args the command line arguments
+     */
+    public static void main(String args[]) {
+        /* Set the Nimbus look and feel */
+        //<editor-fold defaultstate="collapsed" desc=" Look and feel setting code (optional) ">
+        /* If Nimbus (introduced in Java SE 6) is not available, stay with the default look and feel.
+         * For details see http://download.oracle.com/javase/tutorial/uiswing/lookandfeel/plaf.html 
+         */
+        try {
+            for (javax.swing.UIManager.LookAndFeelInfo info : javax.swing.UIManager.getInstalledLookAndFeels()) {
+                if ("Nimbus".equals(info.getName())) {
+                    javax.swing.UIManager.setLookAndFeel(info.getClassName());
+                    break;
+                }
+            }
+        } catch (ClassNotFoundException ex) {
+            java.util.logging.Logger.getLogger(surveyWindow.class.getName()).log(java.util.logging.Level.SEVERE, null, ex);
+        } catch (InstantiationException ex) {
+            java.util.logging.Logger.getLogger(surveyWindow.class.getName()).log(java.util.logging.Level.SEVERE, null, ex);
+        } catch (IllegalAccessException ex) {
+            java.util.logging.Logger.getLogger(surveyWindow.class.getName()).log(java.util.logging.Level.SEVERE, null, ex);
+        } catch (javax.swing.UnsupportedLookAndFeelException ex) {
+            java.util.logging.Logger.getLogger(surveyWindow.class.getName()).log(java.util.logging.Level.SEVERE, null, ex);
+        }
+        //</editor-fold>
+        //</editor-fold>
+
+        /* Create and display the form */
+        java.awt.EventQueue.invokeLater(new Runnable() {
+            @Override
+            public void run() {
+
+                new surveyWindow().setVisible(true);
+            }
+        });
+    }
+
+    /**
+     *  Reads the checkboxes
+     * @return  int array filled with 0s or 1s
+     */
+    public int[] getCheckBox() {
+        boolean[] values = new boolean[9];
+
+        values[0] = checkBoxShooters.isSelected();
+        values[1] = checkBoxFighters.isSelected();
+        values[2] = checkBoxPlatformers.isSelected();
+
+        values[3] = checkBoxStrategy.isSelected();
+        values[4] = checkBoxRPG.isSelected();
+        values[5] = checkBoxPuzzle.isSelected();
+
+        values[6] = checkBoxArcade.isSelected();
+        values[7] = checkBoxSports.isSelected();
+        values[8] = checkBoxSports.isSelected();
```

Comment:
You have checkBoxSports twice (missing checkBoxOther).  Your votes for “Other” are not being counted properly.